data/Kubiak_2009/raw/Kubiak_2009_extra_R_code.R

```diff
@@ -0,0 +1,246 @@
+    peak_fruiting_resp_max = str_extract(peak_fruiting_resp, "(?<=-)[0-9]+"),
+    peak_fruiting_seed_min = str_extract(peak_fruiting_seed, "[0-9]+(?=-)"),
+    peak_fruiting_seed_max = str_extract(peak_fruiting_seed, "(?<=-)[0-9]+"),
+    peak_fruiting_unknown_min = str_extract(peak_fruiting_unknown, "[0-9]+(?=-)"),
+    peak_fruiting_unknown_max = str_extract(peak_fruiting_unknown, "(?<=-)[0-9]+"),
+    fire_time_from_fire_to_flowering_min = str_extract(fire_time_from_fire_to_flowering, "[0-9]+(?=-)"),
+    fire_time_from_fire_to_flowering_max = str_extract(fire_time_from_fire_to_flowering, "(?<=-)[0-9]+"),
+    first_flowering_resp = if_else(str_detect(first_flowering_resp, "-"), NA_character_, first_flowering_resp),
+    first_flowering_seed = if_else(str_detect(first_flowering_seed, "-"), NA_character_, first_flowering_seed),
+    first_flowering_unknown = if_else(str_detect(first_flowering_unknown, "-"), NA_character_, first_flowering_unknown),
+    first_fruiting_resp = if_else(str_detect(first_fruiting_resp, "-"), NA_character_, first_fruiting_resp),
+    first_fruiting_seed = if_else(str_detect(first_fruiting_seed, "-"), NA_character_, first_fruiting_seed),
+    primary_juvenile_period = if_else(str_detect(primary_juvenile_period, "-"), NA_character_, primary_juvenile_period),
+    secondary_juvenile_period = if_else(str_detect(secondary_juvenile_period, "-"), NA_character_, secondary_juvenile_period),
+    peak_flowering_resp = if_else(str_detect(peak_flowering_resp, "-"), NA_character_, peak_flowering_resp),
+    peak_flowering_seed = if_else(str_detect(peak_flowering_seed, "-"), NA_character_, peak_flowering_seed),
+    peak_flowering_unknown = if_else(str_detect(peak_flowering_unknown, "-"), NA_character_, peak_flowering_unknown),
+    peak_fruiting_resp = if_else(str_detect(peak_fruiting_resp, "-"), NA_character_, peak_fruiting_resp),
+    peak_fruiting_seed = if_else(str_detect(peak_fruiting_seed, "-"), NA_character_, peak_fruiting_seed),
+    peak_fruiting_unknown = if_else(str_detect(peak_fruiting_unknown, "-"), NA_character_, peak_fruiting_unknown),
+    fire_time_from_fire_to_flowering = if_else(str_detect(fire_time_from_fire_to_flowering, "-"), NA_character_, fire_time_from_fire_to_flowering)
+  ) -> wide_data
+
+# Changes units from years to weeks (a few exceptions, beware)
+wide_data %>% 
+  mutate(across(c(4:19, 24:55), as.numeric)) %>% # Convert to numeric columns
+  mutate(
+    first_flowering_resp_min = first_flowering_resp_min*52,
+    first_flowering_resp_max = first_flowering_resp_max*52,
+    first_flowering_seed = if_else(first_flowering_seed < 5, first_flowering_seed*52, first_flowering_seed),
+    first_flowering_seed_min = first_flowering_seed_min*52,
+    first_flowering_seed_max = first_flowering_seed_max*52,
+    first_flowering_unknown_min = if_else(first_flowering_unknown_min < 10, first_flowering_unknown_min*52, first_flowering_unknown_min),
```

Comment:
I like how you've created separate min & max traits to keep track of value type, while having a single column of information to think about